drivers/amplifiers/ada4250/ada4250.c

```diff
@@ -463,13 +484,10 @@ int32_t ada4250_set_normal_mode(struct ada4250_dev *dev, bool reconfig)
 			ret = ada4250_set_config(dev);
 			if (ret != SUCCESS)
 				return ret;
-		} else { /* Reset driver configurations */
-			dev->gain = ADA4250_GAIN_1;
-			dev->refbuf_en = false;
-			dev->power_mode = ADA4250_POWER_NORMAL;
-			dev->bias = ADA4250_BIAS_DISABLE;
-			dev->offset_range = ADA4250_RANGE1;
-			dev->bandwidth = ADA4250_BANDWIDTH_LOW;
+		} else { /* Reset device configurations */
+			ret = ada4250_soft_reset(dev);
```

Comment:
Question: are we sure we want to do a software reset here? Before it was only a handler update, now it's actually a device command being run.

Reply:
The handler should reflect the actual device parameter and only updating the handler would not not necessarily be in sync with the actual device parameters.